raiden/blockchain/abi.py

```diff
@@ -115,63 +91,74 @@ def contract_checksum(contract_path):
-    get_contract_path('EndpointRegistry.sol'),
-    'EndpointRegistry',
-    combined='abi',
-)
-
-netting_channel_compiled = get_static_or_compile(
-    get_contract_path('NettingChannelContract.sol'),
-    'NettingChannelContract',
-    combined='abi',
-)
-
-registry_compiled = get_static_or_compile(
-    get_contract_path('Registry.sol'),
-    'Registry',
-    combined='abi',
-)
-
-# pylint: enable=invalid-name
-
-HUMAN_TOKEN_ABI = human_token_compiled['abi']
-CHANNEL_MANAGER_ABI = channel_manager_compiled['abi']
-NETTING_CHANNEL_ABI = netting_channel_compiled['abi']
-REGISTRY_ABI = registry_compiled['abi']
-ENDPOINT_REGISTRY_ABI = endpoint_registry_compiled['abi']
-
-TOKENADDED_EVENT = get_event(REGISTRY_ABI, 'TokenAdded')
-TOKENADDED_EVENTID = event_id(*get_eventname_types(TOKENADDED_EVENT))
-
-CHANNELNEW_EVENT = get_event(CHANNEL_MANAGER_ABI, 'ChannelNew')
-CHANNELNEW_EVENTID = event_id(*get_eventname_types(CHANNELNEW_EVENT))
-
-CHANNELNEWBALANCE_EVENT = get_event(NETTING_CHANNEL_ABI, 'ChannelNewBalance')
-CHANNELNEWBALANCE_EVENTID = event_id(*get_eventname_types(CHANNELNEWBALANCE_EVENT))
-
-CHANNELCLOSED_EVENT = get_event(NETTING_CHANNEL_ABI, 'ChannelClosed')
-CHANNELCLOSED_EVENTID = event_id(*get_eventname_types(CHANNELCLOSED_EVENT))
-
-CHANNELSECRETREVEALED_EVENT = get_event(NETTING_CHANNEL_ABI, 'ChannelSecretRevealed')
-CHANNELSECRETREVEALED_EVENTID = event_id(*get_eventname_types(CHANNELSECRETREVEALED_EVENT))
-
-CHANNELSETTLED_EVENT = get_event(NETTING_CHANNEL_ABI, 'ChannelSettled')
-CHANNELSETTLED_EVENTID = event_id(*get_eventname_types(CHANNELSETTLED_EVENT))
-
-REGISTRY_TRANSLATOR = ContractTranslator(REGISTRY_ABI)
-CHANNEL_MANAGER_TRANSLATOR = ContractTranslator(CHANNEL_MANAGER_ABI)
-NETTING_CHANNEL_TRANSLATOR = ContractTranslator(NETTING_CHANNEL_ABI)
+class ContractManager():
+
+    def __init__(self):
+        self.is_instantiated = False
+
+    def instantiate(self):
+        self.human_standard_token_compiled = get_static_or_compile(
+            get_contract_path('HumanStandardToken.sol'),
+            'HumanStandardToken',
+            combined='abi',
+        )
+
+        self.channel_manager_compiled = get_static_or_compile(
+            get_contract_path('ChannelManagerContract.sol'),
+            'ChannelManagerContract',
+            combined='abi',
+        )
+
+        self.endpoint_registry_compiled = get_static_or_compile(
+            get_contract_path('EndpointRegistry.sol'),
+            'EndpointRegistry',
+            combined='abi',
+        )
+
+        self.netting_channel_compiled = get_static_or_compile(
+            get_contract_path('NettingChannelContract.sol'),
+            'NettingChannelContract',
+            combined='abi',
+        )
+
+        self.registry_compiled = get_static_or_compile(
+            get_contract_path('Registry.sol'),
+            'Registry',
+            combined='abi',
+        )
+
+        self.is_instantiated = True
+
+    def get_abi(self, contract_name):
```

Comment:
Wouldn't it be less error prone to use constants (er even an enum) for the contract names instead of having to type the string name for every invocation?